### PR TITLE
Migrate the "Kill Horde Master" mission to JSON

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Infrastructure "Move the 'Kill Horde Master' mission to JSON formatting"

#### Purpose of change
Migrate things from missionstart.cpp into a more accessible JSON format

#### Describe the solution
I removed the C++ functions from the mission code and replaced it with JSON.

#### Describe alternatives you've considered
Leaving it as is.

#### Testing
I spawned an NPC, took the mission, killed the horde master, and got $2,500 of credit.

#### Additional context
I don't know how to replicate the multiple placement locations in JSON, so I decided on a widely available building for placement purposes.

This is also my first time working with C++. If it comes up and I don't understand what you're talking about, I would like to apologize for my ignorance.

### CODE REVIEW

File: data/json/npcs/missiondef.json [hunk 1]
```diff
@@ -459,7 +459,23 @@
     "difficulty": 5,
     "value": 250000,
     "urgent": true,
-    "start": "kill_horde_master",
+    "start": {
+      "update_mapgen": {
+        "place_monster": [
+          { "monster": "mon_zombie_master", "x": [ 12, 15 ], "y": [ 12, 15 ], "target": true },
```
Comment (Night-Pryanik) [inline data/json/npcs/missiondef.json:465]:
```suggestion
          { "monster": "mon_zombie_master", "x": [ 12, 15 ], "y": [ 12, 15 ], "target": true, "name": "Demonic Soul" },
```

**Review by LyleSY — APPROVED** [review]
This all looks right to me

**Comment by cathalpern** [comment]
Minor - I'd probably do om_terrain : "office_tower" and add om_replace:"hospital"
Since those are usually more crowded, and placing this mission in a house rather than a more crowded location makes it easier

**Review by Maleclypse — APPROVED** [review]

**Comment by Maleclypse** [comment]
> Minor - I'd probably do om_terrain : "office_tower" and add om_replace:"school" or hotel Since those are usually more crowded, and placing this mission in a house rather than a more crowded location makes it easier

I agree with this but am good either way

**Review by Maleclypse — APPROVED** [review]

**Pull request merged**